Update the guides to refer to the io.quarkus.platform BOMs instead of the universe BOM (#19530)
quarkusio/quarkus: docs/src/main/asciidoc/building-my-first-extension.adoc

*Quarkus Platform*

Quarkus platform is a set of extensions that target the primary use-cases of Quarkus as a development framework and can safely be used in any combination in the same application without creating a dependency conflict.

> From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.

Reviewer: ```suggestion
From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order without introducing a dependency conflict.
```

> From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were aligned globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.

Reviewer 2: ```suggestion
From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.
```

*Quarkiverse Hub*

link:https://github.com/quarkiverse[Quarkiverse Hub] is the GitHub organization that provides repository hosting (including build, CI and release publishing setup) for Quarkus extension projects contributed by the community.

> In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can discover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.

Reviewer: ```suggestion
In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem so that the Quarkus community can discover it using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```
would be better? Otherwise it's hard to follow the sentence.

> In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can disover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.

Reviewer 2: ```suggestion
In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can discover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```

- Configure your extension to use the link:https://github.com/quarkusio/quarkus-ecosystem-ci#quarkus-ecosystem-ci[Quarkus Ecosystem CI] to build against the latest Quarkus Core changes;
- Give you the freedom to manage the project and release to Maven Central as you like.

> NOTE: Extensions hosted in the Quarkiverse Hub may or may not end up in the Quarkus platform.

Reviewer 2: Is there any process someone could follow to have a Quarkiverse extension added to the platform?

Author: This would have to be discussed with the platform maintainers. Specifically, the extension should fit the primary purposes/use-cases the platform is targeting and match the quality requirements, i.e. be supersonic, subatomic and work in native.

Reviewer 2: Right, just wondering if that would be something that readers might ask frequently, maybe we should elaborate more this note at some point (doesn't need to be in this PR, ofc)
